metaflow/metaflow_git.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+"""Get git repository information for the package
+
+Functions to retrieve git repository details like URL, branch name, 
+and commit SHA for Metaflow code provenance tracking.
+"""
+
+import os
+import subprocess
+from os import path, name, environ
+from typing import Dict, Union
+
+# Cache for git information to avoid repeated subprocess calls
+_git_info_cache = None
+
+__all__ = ("get_repository_info",)
+
+GIT_COMMAND = "git"
+
```

Comment:
We should refactor to have a single `_run_git_command` command that is used by all the other functions.